new_try.py

```diff
@@ -0,0 +1,203 @@
+fft_padding=31
+padding=fft_padding
+#Calculate Sampling interval in X and Y directions——计算X和Y方向的采样间隔
+
+delta_x = list(delta_x)
+delta_x=sorted(delta_x)
+number_of_samples_x = len(delta_x)
+delta_x = abs(delta_x[1]-delta_x[2])
+
+delta_y = list(delta_y)
+delta_y=sorted(delta_y)
+number_of_samples_y = len(delta_y)
+delta_y = abs(delta_y[1]-delta_y[2])
+#% Scanner Plane Dimensions
+length_x = delta_x * (number_of_samples_x-1)
+length_y = delta_y * (number_of_samples_y-1)
+
+lambda0=299792458/f
+k0=2*np.pi/lambda0
+
+number_of_samples_x_padded = padding*number_of_samples_x
+number_of_samples_y_padded = padding*number_of_samples_y
+
+
+m=np.linspace(-1*number_of_samples_x_padded/2,number_of_samples_x_padded/2-1,number_of_samples_x_padded)
+n=np.linspace(-1*number_of_samples_y_padded/2,number_of_samples_y_padded/2-1,number_of_samples_y_padded)
+
+kx=2*np.pi*m/(number_of_samples_x_padded*delta_x);
+ky=2*np.pi*n/(number_of_samples_y_padded*delta_y);
+
+
+
+kx_grid,ky_grid = np.meshgrid(kx,ky);
+
+ky_grid=np.flipud(ky_grid)
+
+
+ky_grid1=np.flipud(ky_grid)
+d=np.power(ky_grid1,2)
+c=k0*k0-np.power(kx_grid,2)-np.power(ky_grid1,2)
+c=np.array(c,dtype=np.complex)
+kz_grid = np.sqrt(c)
+
+# herical Far-Field Wavenumber Vector——球面远场波数向量kx_grid_spherical+ky_grid_spherical+kz_grid_spherical
+
+
+(theta_grid,phi_grid)=np.meshgrid(theta_range2,phi_range2)
+
+#(theta_grid,phi_grid)=theta_range2,phi_range2
+kx_grid_spherical = k0*np.sin(theta_grid)*np.cos(phi_grid)
+ky_grid_spherical = k0*np.sin(theta_grid)*np.sin(phi_grid)
+kz_grid_spherical = k0*np.cos(theta_grid)
+
+
+#Reshape E field to fit grid——变换得到远场Ax,Ay,Az分量
+
+Ex_nf = Ex.reshape(number_of_samples_y,number_of_samples_x).transpose()
+Ey_nf = Ey.reshape(number_of_samples_y,number_of_samples_x).transpose()
+Ez_nf = Ez.reshape(number_of_samples_y,number_of_samples_x).transpose()
+
+number_of_samples_x2=number_of_samples_x
+number_of_samples_y2=number_of_samples_y
+
+Ex_nf=Ex_nf.transpose()
+Ey_nf=Ey_nf.transpose()
+Ez_nf=Ez_nf.transpose()
+
+Ex_nf=np.flipud(Ex_nf)
+Ex_nf_Interpolation=Ex_nf
+
+Ey_nf=np.flipud(Ey_nf)
+Ey_nf_Interpolation=Ey_nf
+
+Ez_nf=np.flipud(Ez_nf)
+Ez_nf_Interpolation=Ez_nf
+#% 对29*12的平面时域信号Ex_nf和Ey_nf进行补零，形成841*348的矩阵
+
+fx=(ifft2(Ex_nf,(number_of_samples_y_padded,number_of_samples_x_padded)))
+fy=(ifft2(Ey_nf,(number_of_samples_y_padded,number_of_samples_x_padded)))
+ky_grid=np.flipud(ky_grid)
+fz=-(fx*kx_grid+fy*ky_grid)/kz_grid
+
+
+
+fx=ifftshift(fx)
+fy=ifftshift(fy)
+
+fz1=ifftshift(fz)
+
+
+fz2=-(fx*kx_grid+fy*ky_grid)/kz_grid
+fz=fz1*0.65+fz2*0.35
+
+#;%采样平面距离天线口面的距离为5*lambda
+r=5*lambda0
+
+print("here")
+
+
+fx=fx*np.exp(1j*kz_grid*r)/(4*np.pi*np.pi)
+fy=fy*np.exp(1j*kz_grid*r)/(4*np.pi*np.pi)
+fz=fz*np.exp(1j*kz_grid*r)/(4*np.pi*np.pi)
+
+
+
+def yaml_load_noncompliant():
+    import json
+    import yaml
+    response = yaml.dump({'a': 1, 'b': 2, 'c': 3})
+    # Noncompliant: uses unsafe yaml load.
+    result = yaml.load(response)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load(). https://bandit.readthedocs.io/en/latest/plugins/b506_yaml_load.html